gluoncv/model_zoo/residual_attentionnet.py

```diff
@@ -0,0 +1,860 @@
+                self.softmax2_blocks.add(nn.BatchNorm())
+                self.softmax2_blocks.add(nn.Activation('relu'))
+                self.softmax2_blocks.add(nn.Conv2D(channels, kernel_size=1, strides=1, use_bias=False))
+                self.softmax2_blocks.add(nn.Activation('sigmoid'))
+
+            self.last_blocks = ResidualBlock(channels)
+
+    def hybrid_forward(self, F, x, *args, **kwargs):
+        x = self.first_residual_blocks(x)
+        out_trunk = self.trunk_branches(x)
+        out_mpool1 = self.mpool1(x)
+        out_softmax1 = self.softmax1_blocks(out_mpool1)
+        out_interp1 = F.elemwise_add(self.interpolation1(out_softmax1), out_trunk)
+        out_softmax2 = self.softmax2_blocks(out_interp1)
+        out = F.elemwise_add(F.ones_like(out_softmax2), out_softmax2)
+        out = F.elemwise_mul(out, out_trunk)
+        out_last = self.last_blocks(out)
+        return out_last
+
+
+class AttentionModule_stage3_cifar(nn.HybridBlock):
+    def __init__(self, channels, **kwargs):
+        """
+        Input size is 14 x 14
+        :param channels:
+        :param kwargs:
+        """
+        super(AttentionModule_stage3_cifar, self).__init__(**kwargs)
+        with self.name_scope():
+            self.first_residual_blocks = ResidualBlock(channels)
+
+            self.trunk_branches = nn.HybridSequential()
+            with self.trunk_branches.name_scope():
+                self.trunk_branches.add(ResidualBlock(channels))
+                self.trunk_branches.add(ResidualBlock(channels))
+
+            self.softmax1_blocks = nn.HybridSequential()
+            with self.softmax1_blocks.name_scope():
+                self.softmax1_blocks.add(ResidualBlock(channels))
+                self.softmax1_blocks.add(ResidualBlock(channels))
+
+            self.softmax2_blocks = nn.HybridSequential()
+            with self.softmax2_blocks.name_scope():
+                self.softmax2_blocks.add(nn.BatchNorm())
+                self.softmax2_blocks.add(nn.Activation('relu'))
+                self.softmax2_blocks.add(nn.Conv2D(channels, kernel_size=1, strides=1, use_bias=False))
+                self.softmax2_blocks.add(nn.BatchNorm())
+                self.softmax2_blocks.add(nn.Activation('relu'))
+                self.softmax2_blocks.add(nn.Conv2D(channels, kernel_size=1, strides=1, use_bias=False))
+                self.softmax2_blocks.add(nn.Activation('sigmoid'))
+
+            self.last_blocks = ResidualBlock(channels)
+
+    def hybrid_forward(self, F, x, *args, **kwargs):
+        x = self.first_residual_blocks(x)
+        out_trunk = self.trunk_branches(x)
+        out_softmax1 = self.softmax1_blocks(x)
+        out_softmax2 = self.softmax2_blocks(out_softmax1)
+        out = F.elemwise_add(F.ones_like(out_softmax2), out_softmax2)
+        out = F.elemwise_mul(out, out_trunk)
+        out_last = self.last_blocks(out)
+
+        return out_last
+
+
+class ResidualAttentionModel_448input(nn.HybridBlock):
```

Comment:
Please try the best to parameterize the difference among the networks, which means that we only need one definition of the network, and the difference (for example the number of channels) can be input arguments.

One example: https://github.com/hetong007/gluon-cv/blob/master/gluoncv/model_zoo/cifarresnet.py#L245

Reply:
OK,I'll refactor code.